### ISSUE TITLE
Crash (SIGSEGV) PodYandexMobileMetrica_ksthread_getQueueName

### ISSUE THREAD

**Issue by captainuberawesome** [yandexmobile/metrica-sdk-ios#70]
Hello,

We have received several crash reports with the same crash (happened to 2 users 10 times in the last 24 hours). All crashes happened on iPhone 8 Plus running iOS 11.3.1 (app in foreground). Here's the crash log from Fabric:

```
Date: 2018-05-29T10:54:00Z
OS Version: 11.3.1 (15E302)
Device: iPhone 8 Plus
RAM Free: 4%
Disk Free: 73.3%

#3. Crashed: Thread
0  YandexMobileMetrica            0x1033d1774 PodYandexMobileMetrica_ksthread_getQueueName + 128
1  YandexMobileMetrica            0x1033d175c PodYandexMobileMetrica_ksthread_getQueueName + 104
2  YandexMobileMetrica            0x1033ba478 PodYandexMobileMetrica_ksccd_init + 488
3  libsystem_pthread.dylib        0x1842f5220 _pthread_body + 272
4  libsystem_pthread.dylib        0x1842f5110 _pthread_body + 290
5  libsystem_pthread.dylib        0x1842f3b10 thread_start + 4

--

#0. com.apple.main-thread
0  libsystem_kernel.dylib         0x184133e08 mach_msg_trap + 8
1  libsystem_kernel.dylib         0x184133c80 mach_msg + 72
2  QuartzCore                     0x1887b3620 CA::Context::synchronize_transactions() + 320
3  UIKit                          0x18e68680c _runAfterCACommitDeferredBlocks + 124
4  UIKit                          0x18e67c98c _cleanUpAfterCAFlushAndRunDeferredBlocks + 384
5  UIKit                          0x18e68d6c0 __34-[UIApplication _firstCommitBlock]_block_invoke_2 + 152
6  CoreFoundation                 0x1846772bc __CFRUNLOOP_IS_CALLING_OUT_TO_A_BLOCK__ + 20
7  CoreFoundation                 0x184676a7c __CFRunLoopDoBlocks + 264
8  CoreFoundation                 0x1846747b0 __CFRunLoopRun + 1224
9  CoreFoundation                 0x184594da8 CFRunLoopRunSpecific + 552
10 GraphicsServices               0x186577020 GSEventRunModal + 100
11 UIKit                          0x18e57578c UIApplicationMain + 236
12 AppName                        0x1025c5dcc main (PartyTicketsView.swift:28)
13 libdyld.dylib                  0x184025fc0 start + 4

#1. com.apple.uikit.eventfetch-thread
0  libsystem_kernel.dylib         0x184133e08 mach_msg_trap + 8
1  libsystem_kernel.dylib         0x184133c80 mach_msg + 72
2  CoreFoundation                 0x184676e40 __CFRunLoopServiceMachPort + 196
3  CoreFoundation                 0x184674908 __CFRunLoopRun + 1568
4  CoreFoundation                 0x184594da8 CFRunLoopRunSpecific + 552
5  Foundation                     0x185009674 -[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 304
6  Foundation                     0x18500951c -[NSRunLoop(NSRunLoop) runUntilDate:] + 148
7  UIKit                          0x18e25a7e4 -[UIEventFetcher threadMain] + 136
8  Foundation                     0x185119efc __NSThread__start__ + 1040
9  libsystem_pthread.dylib        0x1842f5220 _pthread_body + 272
10 libsystem_pthread.dylib        0x1842f5110 _pthread_body + 290
11 libsystem_pthread.dylib        0x1842f3b10 thread_start + 4

#2. com.twitter.crashlytics.ios.MachExceptionServer
0  libsystem_kernel.dylib         0x184133e08 mach_msg_trap + 8
1  libsystem_kernel.dylib         0x184133c80 mach_msg + 72
2  AppName                        0x1027d8334 CLSMachExceptionServer (CLSMachException.c:180)
3  libsystem_pthread.dylib        0x1842f5220 _pthread_body + 272
4  libsystem_pthread.dylib        0x1842f5110 _pthread_body + 290
5  libsystem_pthread.dylib        0x1842f3b10 thread_start + 4

#3. Crashed: Thread
0  YandexMobileMetrica            0x1033d1774 PodYandexMobileMetrica_ksthread_getQueueName + 128
1  YandexMobileMetrica            0x1033d175c PodYandexMobileMetrica_ksthread_getQueueName + 104
2  YandexMobileMetrica            0x1033ba478 PodYandexMobileMetrica_ksccd_init + 488
3  libsystem_pthread.dylib        0x1842f5220 _pthread_body + 272
4  libsystem_pthread.dylib        0x1842f5110 _pthread_body + 290
5  libsystem_pthread.dylib        0x1842f3b10 thread_start + 4

#4. com.apple.NSURLConnectionLoader
0  libsystem_kernel.dylib         0x184133e08 mach_msg_trap + 8
1  libsystem_kernel.dylib         0x184133c80 mach_msg + 72
2  CoreFoundation                 0x184676e40 __CFRunLoopServiceMachPort + 196
3  CoreFoundation                 0x184674908 __CFRunLoopRun + 1568
4  CoreFoundation                 0x184594da8 CFRunLoopRunSpecific + 552
5  CFNetwork                      0x184ea24b0 -[__CoreSchedulingSetRunnable runForever] + 220
6  Foundation                     0x185119efc __NSThread__start__ + 1040
7  libsystem_pthread.dylib        0x1842f5220 _pthread_body + 272
8  libsystem_pthread.dylib        0x1842f5110 _pthread_body + 290
9  libsystem_pthread.dylib        0x1842f3b10 thread_start + 4

#5. RLMRealm notification listener
0  libsystem_kernel.dylib         0x18413834c kevent + 8
1  Realm                          0x103515be0 realm::_impl::ExternalCommitHelper::listen() (external_commit_helper.cpp:203)
2  Realm                          0x10351654c std::__1::__async_func<realm::_impl::ExternalCommitHelper::ExternalCommitHelper(realm::_impl::RealmCoordinator&)::$_0>::operator()() (future:2323)
3  Realm                          0x1035164d4 std::__1::__async_assoc_state<void, std::__1::__async_func<realm::_impl::ExternalCommitHelper::ExternalCommitHelper(realm::_impl::RealmCoordinator&)::$_0> >::__execute() (future:1041)
4  Realm                          0x103516684 std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (std::__1::__async_assoc_state<void, std::__1::__async_func<realm::_impl::ExternalCommitHelper::ExternalCommitHelper(realm::_impl::RealmCoordinator&)::$_0> >::*)(), std::__1::__async_assoc_state<void, std::__1::__async_func<realm::_impl::ExternalCommitHelper::ExternalCommitHelper(realm::_impl::RealmCoordinator&)::$_0> >*> >(void*, void*) (memory:2549)
5  libsystem_pthread.dylib        0x1842f5220 _pthread_body + 272
6  libsystem_pthread.dylib        0x1842f5110 _pthread_body + 290
7  libsystem_pthread.dylib        0x1842f3b10 thread_start + 4

#6. Thread
0  libsystem_kernel.dylib         0x1841550f4 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x1842fbed8 _pthread_cond_wait$VARIANT$armv81 + 628
2  libc++.1.dylib                 0x18380c25c std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 24
3  JavaScriptCore                 0x18bceb1a4 void std::__1::condition_variable_any::wait<std::__1::unique_lock<bmalloc::Mutex> >(std::__1::unique_lock<bmalloc::Mutex>&) + 104
4  JavaScriptCore                 0x18bceadb8 bmalloc::Scavenger::threadRunLoop() + 164
5  JavaScriptCore                 0x18bceab60 bmalloc::Scavenger::Scavenger(std::__1::lock_guard<bmalloc::StaticMutex>&) + 10
6  JavaScriptCore                 0x18bceaf7c void* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (*)(bmalloc::Scavenger*), bmalloc::Scavenger*> >(void*) + 44
7  libsystem_pthread.dylib        0x1842f5220 _pthread_body + 272
8  libsystem_pthread.dylib        0x1842f5110 _pthread_body + 290
9  libsystem_pthread.dylib        0x1842f3b10 thread_start + 4

#7. WebThread
0  libsystem_kernel.dylib         0x184133e08 mach_msg_trap + 8
1  libsystem_kernel.dylib         0x184133c80 mach_msg + 72
2  CoreFoundation                 0x184676e40 __CFRunLoopServiceMachPort + 196
3  CoreFoundation                 0x184674908 __CFRunLoopRun + 1568
4  CoreFoundation                 0x184594da8 CFRunLoopRunSpecific + 552
5  WebCore                        0x18c821d9c RunWebThread(void*) + 592
6  libsystem_pthread.dylib        0x1842f5220 _pthread_body + 272
7  libsystem_pthread.dylib        0x1842f5110 _pthread_body + 290
8  libsystem_pthread.dylib        0x1842f3b10 thread_start + 4

#8. VKSdkNetworking
0  libsystem_kernel.dylib         0x184133e08 mach_msg_trap + 8
1  libsystem_kernel.dylib         0x184133c80 mach_msg + 72
2  CoreFoundation                 0x184676e40 __CFRunLoopServiceMachPort + 196
3  CoreFoundation                 0x184674908 __CFRunLoopRun + 1568
4  CoreFoundation                 0x184594da8 CFRunLoopRunSpecific + 552
5  Foundation                     0x185009674 -[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 304
6  Foundation                     0x18507e1a8 -[NSRunLoop(NSRunLoop) run] + 88
7  VK_ios_sdk                     0x1032add48 +[VKHTTPOperation networkRequestThreadEntryPoint:] (VKHTTPOperation.m:89)
8  Foundation                     0x185119efc __NSThread__start__ + 1040
9  libsystem_pthread.dylib        0x1842f5220 _pthread_body + 272
10 libsystem_pthread.dylib        0x1842f5110 _pthread_body + 290
11 libsystem_pthread.dylib        0x1842f3b10 thread_start + 4

#9. Thread
0  libsystem_kernel.dylib         0x184155d84 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x1842f3eb4 _pthread_wqthread + 928
2  libsystem_pthread.dylib        0x1842f3b08 start_wqthread + 4

#10. Thread
0  libsystem_pthread.dylib        0x1842f3b04 start_wqthread + 122

#11. Thread
0  libsystem_kernel.dylib         0x184155d84 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x1842f3eb4 _pthread_wqthread + 928
2  libsystem_pthread.dylib        0x1842f3b08 start_wqthread + 4

#12. io.answers.EventQueue (QOS: BACKGROUND)
0  libsystem_kernel.dylib         0x184154148 __bsdthread_ctl + 8
1  libsystem_pthread.dylib        0x184300734 _pthread_qos_override_end_direct + 32
2  libobjc.A.dylib                0x183891c44 rwlock_tt<false>::unlockRead() + 56
3  libobjc.A.dylib                0x183891b34 lookUpImpOrForward + 512
4  libobjc.A.dylib                0x18389cc38 _objc_msgSend_uncached + 56
5  AppName                        0x1027fcd6c -[ANSBackoffTimer executeBlockAfterNextInterval:] (ANSBackoffTimer.m:49)
6  AppName                        0x1027fd6d4 __111-[ANSNetworkRetryHandler handleCompletedRequest:withResponse:forIdentifier:maxRetries:error:completionHandler:]_block_invoke (ANSNetworkRetryHandler.m:115)
7  AppName                        0x1028130fc +[FABNetworkResponseHandler handleCompletedResponse:forOriginalRequest:error:block:] (FABNetworkResponseHandler.m:232)
8  AppName                        0x1027fd4e0 -[ANSNetworkRetryHandler handleCompletedRequest:withResponse:forIdentifier:maxRetries:error:completionHandler:] (ANSNetworkRetryHandler.m:92)
9  AppName                        0x1027fd3b0 -[ANSNetworkRetryHandler handleCompletedRequest:withResponse:forIdentifier:error:completionHandler:] (ANSNetworkRetryHandler.m:82)
10 AppName                        0x1028085f4 -[ANSAnswersController uploader:completedRequest:withResponse:uniqueName:error:completionHandler:] (ANSAnswersController.m:556)
11 AppName                        0x102802f78 -[ANSUploader URLSession:task:didCompleteWithError:] (ANSUploader.m:353)
12 CFNetwork                      0x184e23760 __51-[NSURLSession delegate_task:didCompleteWithError:]_block_invoke.207 + 76
13 Foundation                     0x1850f6e88 __NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 16
14 Foundation                     0x1850388d0 -[NSBlockOperation main] + 72
15 Foundation                     0x185037cac -[__NSOperationInternal _start:] + 848
16 libdispatch.dylib              0x183fc0ae4 _dispatch_client_callout + 16
17 libdispatch.dylib              0x183ffcb0c _dispatch_block_invoke_direct$VARIANT$armv81 + 216
18 libdispatch.dylib              0x183fc0ae4 _dispatch_client_callout + 16
19 libdispatch.dylib              0x183ffcb0c _dispatch_block_invoke_direct$VARIANT$armv81 + 216
20 libdispatch.dylib              0x183ffca00 dispatch_block_perform$VARIANT$armv81 + 104
21 Foundation                     0x1850f8750 __NSOQSchedule_f + 376
22 libdispatch.dylib              0x183fc0ae4 _dispatch_client_callout + 16
23 libdispatch.dylib              0x183ffd7a8 _dispatch_continuation_pop$VARIANT$armv81 + 416
24 libdispatch.dylib              0x183ffc180 _dispatch_async_redirect_invoke$VARIANT$armv81 + 596
25 libdispatch.dylib              0x1840023a4 _dispatch_root_queue_drain + 592
26 libdispatch.dylib              0x1840020f0 _dispatch_worker_thread3 + 112
27 libsystem_pthread.dylib        0x1842f3fac _pthread_wqthread + 1176
28 libsystem_pthread.dylib        0x1842f3b08 start_wqthread + 4

#13. Thread
0  libsystem_kernel.dylib         0x184155d84 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x1842f3eb4 _pthread_wqthread + 928
2  libsystem_pthread.dylib        0x1842f3b08 start_wqthread + 4

#14. Thread
0  libsystem_kernel.dylib         0x184155d84 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x1842f40a0 _pthread_wqthread + 1420
2  libsystem_pthread.dylib        0x1842f3b08 start_wqthread + 4

#15. com.apple.NSURLSession-work
0  libsystem_kernel.dylib         0x184133e44 semaphore_wait_trap + 8
1  libdispatch.dylib              0x183ff20bc _dispatch_sema4_wait$VARIANT$armv81 + 24
2  libdispatch.dylib              0x183ff2a38 _dispatch_semaphore_wait_slow + 128
3  CFNetwork                      0x184d10604 -[__NSURLBackgroundSession setupBackgroundSession] + 748
4  CFNetwork                      0x184d10300 -[__NSURLBackgroundSession ensureRemoteSession] + 80
5  libdispatch.dylib              0x183fc0b24 _dispatch_call_block_and_release + 24
6  libdispatch.dylib              0x183fc0ae4 _dispatch_client_callout + 16
7  libdispatch.dylib              0x183fff1b4 _dispatch_queue_serial_drain$VARIANT$armv81 + 568
8  libdispatch.dylib              0x183fffad8 _dispatch_queue_invoke$VARIANT$armv81 + 328
9  libdispatch.dylib              0x18400047c _dispatch_root_queue_drain_deferred_wlh$VARIANT$armv81 + 332
10 libdispatch.dylib              0x18400844c _dispatch_workloop_worker_thread$VARIANT$armv81 + 612
11 libsystem_pthread.dylib        0x1842f3e70 _pthread_wqthread + 860
12 libsystem_pthread.dylib        0x1842f3b08 start_wqthread + 4

#16. Thread
0  libsystem_kernel.dylib         0x184155d84 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x1842f3eb4 _pthread_wqthread + 928
2  libsystem_pthread.dylib        0x1842f3b08 start_wqthread + 4

#17. Thread
0  libsystem_kernel.dylib         0x184155d84 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x1842f3eb4 _pthread_wqthread + 928
2  libsystem_pthread.dylib        0x1842f3b08 start_wqthread + 4

#18. Thread
0  libsystem_kernel.dylib         0x184155d84 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x1842f3eb4 _pthread_wqthread + 928
2  libsystem_pthread.dylib        0x1842f3b08 start_wqthread + 4

#19. Thread
0  libsystem_pthread.dylib        0x1842f3b04 start_wqthread + 122

#20. Thread
0  libsystem_kernel.dylib         0x184155d84 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x1842f3eb4 _pthread_wqthread + 928
2  libsystem_pthread.dylib        0x1842f3b08 start_wqthread + 4
```

AppMetrica is installed via CocoaPods
`pod 'YandexMobileMetrica/Dynamic'`

We're only using AppMetrica for basic stats, the only code that uses AppMetrica in the app is the initialisation code:
```
private func initYandex() {
  let apiKey = Constants.appMetricaAPIKey
  guard let configuration = YMMYandexMetricaConfiguration(apiKey: apiKey) else { return }
  YMMYandexMetrica.activate(with: configuration)
}
```

**Comment by bamx23** [comment]
Hi! This problem was fixed in version 3.0.1

**Comment by captainuberawesome** [comment]
@bamx23 

Thank you for your time, I will close the issue then.
